app/services/batched_remove_status_service.rb

```diff
@@ -27,19 +27,14 @@ def call(statuses, **options)
     # transaction lock the database, but we use the delete method instead
     # of destroy to avoid all callbacks. We rely on foreign keys to
     # cascade the delete faster without loading the associations.
-    statuses_and_reblogs.each(&:delete)
+    statuses_and_reblogs.each_slice(50) { |slice| Status.where(id: slice.map(&:id)).delete_all }
```

Comment:
I guess this is fine since an account being deleted would likely not see other processes trying to access or update its statuses anymore. If there is no congestion due to this then the batches might as well be larger, I guess we'll find out in production how high it can be set.